tests/kernel/stack/stack_api/prj.conf

```diff
@@ -2,3 +2,5 @@ CONFIG_ZTEST=y
 CONFIG_IRQ_OFFLOAD=y
 CONFIG_USERSPACE=y
 CONFIG_DYNAMIC_OBJECTS=y
+CONFIG_OBJECT_TRACING=y
+CONFIG_HEAP_MEM_POOL_SIZE=128
```

Comment:
Why is this turned on?
k_stack_alloc_init() uses z_thread_malloc() which does not pull RAM from here, it uses the thread resource pool assigned in test_main

Reply:
@andrewboie  We want to validate k_stack_alloc_init()  api in kernel mode. Thread resource pool assignment is happening  through k_thread_resource_pool_assign() in test_main, when user space is enabled.
We are adding this test to improve code coverage on native posix platform, where user space is not enabled. That is the reason, we are using k_thread_system_pool_assign() and allocating memory from this heap memory pool.